pkgs/applications/science/math/gap/default.nix

```diff
@@ -1,37 +1,81 @@
-{ stdenv, fetchurl, m4, gmp }:
-
-let
-  baseName = "gap";
-  version = "4r8p3";
-
-  pkgVer = "2016_03_19-22_17";
-in
+{ stdenv
+, fetchurl
+, fetchpatch
+, m4
+, gmp
+# don't remove any packages -- results in a ~1.3G size increase
+, keepAllPackages ? false
+}:
 
 stdenv.mkDerivation rec {
-  name = "${baseName}-${version}";
+  pname = "gap";
+  version = "4r8p10";
+  pkgVer = "2018_01_15-13_02";
+  name = "${pname}-${version}";
 
   src = fetchurl {
-    url = "ftp://ftp.gap-system.org/pub/gap/gap48/tar.gz/${baseName}${version}_${pkgVer}.tar.gz";
-    sha256 = "1rmb0lj43avv456sjwb7ia3y0wwk5shlqylpkdwnnqpjnvjbnzv6";
+    # https://www.gap-system.org/Releases/
+    # newer versions (4.9.0) are available, but still considered beta (https://github.com/gap-system/gap/wiki/GAP-4.9-release-notes)
```

Comment:
If you change anything else, this comment should probably go up near the version declaration.